src/ML-METATENSOR/metatensor_system.cpp

```diff
@@ -520,10 +520,23 @@ metatensor_torch::System MetatensorSystemAdaptor::system_from_lmp(
         cell = cell.matmul(model_strain);
     }
 
+    // periodic boundary conditions handling:
+    // for now we assume that the system is fully periodic and we throw an error otherwise
+    if (!domain->xperiodic || !domain->yperiodic || !domain->zperiodic) {
+        error->all(FLERR, "pair_metatensor requires a fully periodic system");
+    }
+    auto pbc = torch::tensor({domain->xperiodic, domain->yperiodic, domain->zperiodic}, torch::TensorOptions().dtype(torch::kBool).device(device));
+    // note that something like this:
+    // cell.index_put_({torch::logical_not(pbc)}, torch::tensor({0.0}, torch::TensorOptions().dtype(dtype).device(device)));
+    // would allow creating System with non-periodic directions, but we're using the inverse of the cell
+    // matrix to filter the neighbor list, and the cell matrix becomes singular if any of its rows are zero.
+    // This requires some changes in the neighbor list filtering code to handle non-periodic directions.
```

Comment:
Could we not use a different matrix for the NL filtering than the one we put in the `System`?

Reply:
Yes, we can! Are you confident our current code works however in cases with mixed PBCs? I can see potential silent issues with the way we use the `torch.round` function, which wouldn't warn when we're not close to an integer.

More in general, I think the correct design for `System` is not to check whether a False PBC corresponds to a row of zeros. Both ASE and LAMMPS seem to support that case and just ignore the cell row (or use it to calculate the cell volume and related quantities that don't influence MD). Moreover, this check will always cost us a synchronization on GPU, since we have to take the `.item<bool>()` of a GPU tensor to perform the check. I can open an issue about this in metatensor if you want

Reply:
Actually, if pbc is disabled in a direction, the cell shift in this direction will always be 0, so we don't have to compute it.

The current code might not work with mixed PBC, but we should be able to add a couple `if` in there to check pbc.

Reply:
> use it to calculate the cell volume 

The cell volume is not well defined if you don't have three periodic dimensions.